
##<b> Since 2008, guests and hosts have used Airbnb to expand on traveling possibilities and present a more unique, personalized way of experiencing the world. Today, Airbnb became one of a kind service that is used and recognized by the whole world. Data analysis on millions of listings provided through Airbnb is a crucial factor for the company. These millions of listings generate a lot of data - data that can be analyzed and used for security, business decisions, understanding of customers' and providers' (hosts) behavior and performance on the platform, guiding marketing initiatives, implementation of innovative additional services and much more. </b>

## <b>This dataset has around 49,000 observations in it with 16 columns and it is a mix between categorical and numeric values. </b>

## <b> Explore and analyze the data to discover key understandings (not limited to these) such as : 
* What can we learn about different hosts and areas?
* What can we learn from predictions? (ex: locations, prices, reviews, etc)
* Which property type is the busiest and why?
* Is there any noticeable difference of traffic among different areas and what could be the reason for it? </b>

We have analyze the given dataset well and We will not only findout the answers of the above questions but we will also find the answer of these questions -

*      Which type of room has the highest and lowest reviews and why?
*      Is there any other crucial factor for hosts than pricing? 

To get the answers of these questions, We're going to findout their most relevant answers.


Now we're going to explore it to understand the key understandings of the Dataset

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from google.colab import drive
%matplotlib inline

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Airbnb NYC 2019.csv')

In [ ]:
df.head()

id  ... availability_365
0  2539  ...              365
1  2595  ...              355
2  3647  ...              365
3  3831  ...              194
4  5022  ...                0

[5 rows x 16 columns]

In [ ]:
df.shape

(48895, 16)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [ ]:
df.describe()

id  ...  availability_365
count  4.889500e+04  ...      48895.000000
mean   1.901714e+07  ...        112.781327
std    1.098311e+07  ...        131.622289
min    2.539000e+03  ...          0.000000
25%    9.471945e+06  ...          0.000000
50%    1.967728e+07  ...         45.000000
75%    2.915218e+07  ...        227.000000
max    3.648724e+07  ...        365.000000

[8 rows x 10 columns]

# 1 - Data Cleaning


Our data has almost 48895 rows for 16 columns. These columns are important to understand the key understandings and to get answers for our questions but also some of these columns have not impact on the insights generated by these dataset.

So, we will drop these columns from the data but, We have to analyze each and every columns first so that we can findout which columns to drop.

In [ ]:
df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')

After analyzing the dataset, We have got to know that every column is equally important for the most relevant insights from our dataset.




In [ ]:
df.head()

id  ... availability_365
0  2539  ...              365
1  2595  ...              355
2  3647  ...              365
3  3831  ...              194
4  5022  ...                0

[5 rows x 16 columns]

As our dataset have almost ~49000 rows, It may possible to have some missing or wrong data in our dataset. Missing data and wrong data will definitely impact our key insights that we'll generate from our dataset. 

So, We'll check for the null values and wrong values in our dataset and drop them.

## Missing values

In [ ]:
(df.isnull().sum() * 100) / df.shape[0]

id                                 0.000000
name                               0.032723
host_id                            0.000000
host_name                          0.042949
neighbourhood_group                0.000000
neighbourhood                      0.000000
latitude                           0.000000
longitude                          0.000000
room_type                          0.000000
price                              0.000000
minimum_nights                     0.000000
number_of_reviews                  0.000000
last_review                       20.558339
reviews_per_month                 20.558339
calculated_host_listings_count     0.000000
availability_365                   0.000000
dtype: float64

After analyzing the dataset, We got the following null rows for columns - 


1.   **0.0328%** data is missing for [name] column
2.   **20.56%** data is missing for [last_review] column
3.   **20.56%** data is missing for [reviews_per_month] column

A small percentage of data is missing for **[name]** column but we will completely ignore that data as properties name will not much impact on the key insights of our dataset.

While more than **20 percent** of data is missing for **[last_review , reviews_per_month]** column.

**20.55%** is much more data so we can not drop these rows as it will impact the key analysis of our dataset.

These two columns are dependent on each other because, If a property has atleast one review then that review's last date will show on the first column.

To take action at these null values, We had take a deep look at these columns data. 

After analyzing the data, What I'm thinking about these null values is, These are the reviews rows. So some of the property might be listed new who doesn't got any review till now.

So, Null value doesn't mean it is wrong data. It may be right data as these properties may also be new who doesn't have any single review now.

There were **two unique ID's** available in our dataframe so we will remove these ID as our dataset already have index no which again providing the unique value for the properties.

We're also removing the **last_review column** as it is not that much important for finding the insights. 

In [ ]:
df= df.drop(['last_review', 'host_id', 'id'],axis=1)

In [ ]:
# review_per_month have null values we can replace it with 0
df['reviews_per_month']=df['reviews_per_month'].replace(to_replace=np.nan,value=0).astype('int64')

In [ ]:
df.dtypes

name                               object
host_name                          object
neighbourhood_group                object
neighbourhood                      object
latitude                          float64
longitude                         float64
room_type                          object
price                               int64
minimum_nights                      int64
number_of_reviews                   int64
reviews_per_month                   int64
calculated_host_listings_count      int64
availability_365                    int64
dtype: object

## Checking Price feature For Outliers/Wrong Data

Price is the most crucial factor for our dataset. Different properties have different pricing. They may variate in pricing from location to location.

But no one wanted to list there properties at **0 cost** (Atleast we're assuming this)

So, We're checking the price of the property. If any property price = 0 then we assume that these entries are wrong and we will remove that rows.

In [ ]:
df[df['price'] == 0]

name  ... availability_365
23161  Huge Brooklyn Brownstone Living, Close to it all.  ...               28
25433      ★Hostel Style Room | Ideal Traveling Buddies★  ...              127
25634    MARTIAL LOFT 3: REDEMPTION (upstairs, 2nd room)  ...                0
25753                    Sunny, Quiet Room in Greenpoint  ...                0
25778      Modern apartment in the heart of Williamsburg  ...               73
25794  Spacious comfortable master bedroom with nice ...  ...              176
25795  Contemporary bedroom in brownstone with nice view  ...              232
25796       Cozy yet spacious private brownstone bedroom  ...              222
26259                              the best you can find  ...                0
26841  Coliving in Brooklyn! Modern design / Shared room  ...              333
26866             Best Coliving space ever! Shared room.  ...              139

[11 rows x 13 columns]

In [ ]:
# Dropping the rows where price = 0

temp_rows = df.index[df['price'] == 0].tolist()

In [ ]:
df.drop(temp_rows, inplace=True)

In [ ]:
df.shape

(48884, 13)